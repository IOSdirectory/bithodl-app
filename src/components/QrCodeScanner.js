import React, { Component } from 'react';
import { StyleSheet, View, Image, Dimensions } from 'react-native';
import PropTypes from 'prop-types';
import Camera from 'react-native-camera';

const windowHeight = Dimensions.get('window').height;
const windowWidth = Dimensions.get('window').width;

const styles = StyleSheet.create({
  view: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#ffffff',
    position: 'absolute',
    top: 0,
    left: 0
  },
  camera: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    height: windowHeight,
    width: windowWidth
  },
  overlay: {
    flex: 1,
    alignSelf: 'stretch',
    backgroundColor: 'rgba(255,255,255,0.5)',
    opacity: 0.7
  },
  viewport: {
    width: 200,
    height: 200,
    position: 'absolute',
    opacity: 0
  }
});

export default class QrCodeScanner extends Component {
  render() {
    const overlayStyles = [
      styles.overlay,
      { opacity: this.props.opacity }
    ];

    const viewportStyles = [
      styles.viewport,
      { opacity: (1 - this.props.opacity) / 4 }
    ];

    return (
      <View style={styles.view}>
        <Camera style={styles.camera} onBarCodeRead={this.props.onScan} keepAwake={true}>
          <View style={overlayStyles} />
          <Image source={require('../img/qr-viewport.png')} style={viewportStyles} />
        </Camera>
      </View>
    );
  }
}

QrCodeScanner.propTypes = {
  opacity: PropTypes.number,
  onScan: PropTypes.func.isRequired
};
