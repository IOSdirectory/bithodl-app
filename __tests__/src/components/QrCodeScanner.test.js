import React from 'react';
import renderer from 'react-test-renderer';
import QrCodeScanner from '../../../src/components/QrCodeScanner';

describe('QrCodeScanner', () => {
  it('renders correctly', () => {
    const onScan = jest.fn();

    const tree = renderer.create(
      <QrCodeScanner opacity={0} onScan={onScan} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
