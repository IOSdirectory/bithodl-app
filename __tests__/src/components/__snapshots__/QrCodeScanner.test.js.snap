// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QrCodeScanner renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
    }
  }
>
  <View
    keepAwake={true}
    onBarCodeRead={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 1334,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "rgba(255,255,255,0.5)",
            "flex": 1,
            "opacity": 0.7,
          },
          Object {
            "opacity": 0,
          },
        ]
      }
    />
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 200,
            "opacity": 0,
            "position": "absolute",
            "width": 200,
          },
          Object {
            "opacity": 0.25,
          },
        ]
      }
    />
  </View>
</View>
`;
